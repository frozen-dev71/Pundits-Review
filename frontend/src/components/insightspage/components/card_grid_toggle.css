/*
// ======================================
// CSS for CARD/GRID toggle menu
// ======================================
*/


.card-grid-toggle-wrapper {
	display: flex;
	justify-content: flex-end;
	height: 1.25em;
	width: 95vw;
}

.card-grid-button-container {
	width: 35%;
	display: flex;
}

.card-grid-button {
	width: 50%;
	display: flex;
	align-items: center;
	justify-content: center;
	background-color: white;
	border: solid black 1px;
	letter-spacing: .2em;
	font-size: .65em;
	cursor: pointer;
}

.card-grid-button-active {
	background-color: black;
	color: white;
}



/*
// ======================================
// MEDIA QUERIES
// ======================================
*/

@media (min-width: 600px) {


    .card-grid-button-container {
		width: 25%;
	}

	.card-grid-toggle-wrapper {
		width: 85vw;
	}

	 .card-grid-button {
        font-size: 0.75em;
    }

}



@media (min-width: 800px) {


    .card-grid-button-container {
		width: 20%;
	}

}


@media (min-width: 1000px) {


    .card-grid-toggle-wrapper {
		width: 75vw;
	}

}


@media (min-width: 1200px) {


    .card-grid-button-container {
		width: 17.5%;
	}

	 .card-grid-button {
        font-size: 0.8em;
    }

}



@media (min-width: 1400px) {


   .card-grid-toggle-wrapper {
		width: 65vw;
	}
}