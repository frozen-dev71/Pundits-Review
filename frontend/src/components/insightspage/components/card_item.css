/*
// ======================================
// CSS for card item component - individual player container (INSIGHTS PAGE)
// ======================================
*/


.insights-card-item-wrapper {
	display: flex;
	flex-direction: column;
	align-items: center;
	height: min-content;
	width: 70vw;
	box-shadow: 4px 4px 10px grey;
	justify-content: space-between;
}

.insights-card-item-header-container {
	display: flex;
	width: 100%;
	justify-content: space-between;
	height: 5%;
	align-items: center;
    padding: .5em 0;
}

.insights-card-item-player-name-header {
	font-size: .65em;
	margin-left: 1em;
	letter-spacing: .25em;
}

.insights-card-item-rank-header {
	font-size: .65em;
	margin-right: 1.5em;
	letter-spacing: .25em;
}

.insights-card-item-player-image-container {
	background-color: white;
	display: grid;
	grid-template-columns: repeat(8,1fr);
    grid-template-rows: repeat(3,1fr);
    height: 20vh;
    min-height: 120px;
    width: 100%;
}


.insights-card-club-badge  {
	grid-row: 1/2;
    grid-column: 1/3;
    align-self: center;
    width: 3em;
    height: 3em;
    margin: .1em;
}

.insights-card-player-shirt {
	grid-row: 2/4;
    grid-column: 3/7;
    align-self: flex-end;
    justify-self: center;
    width: 6em;
    height: 6em;
    z-index: 60;
}

.insights-card-player-name-inner {
	grid-row: 2;
    grid-column: 1/9;
    justify-self: center;
    align-self: center;
    color: #EBECED;
    letter-spacing: .25em;
    font-size: 1.5em;
    text-align: center;
    z-index: 50;
    overflow: auto;
}

.insights-card-club-name {
	grid-row: 1/2;
    grid-column: 3/9;
    height: min-content;
    justify-self: flex-end;
    align-self: flex-start;
    margin: 1em;
    font-size: .5em;
    letter-spacing: .2em;
    padding: .5em;
    background-color: black;
}


.insights-card-item-review-score-container {
	width: 100%;
    /* height: 40%; */
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: center;
    padding: .5em 0;

}

.insights-card-item-score-category {
	letter-spacing: .2em;
	font-size: .75em;
}

.insights-card-item-score-category-score  {
	letter-spacing: .25em;
    padding: .5em 0;
}

.insights-card-item-positive-negative-neutral-grid-container {
	display: grid;
	grid-template-columns: repeat(6,1fr);
    grid-template-rows: repeat(4,1fr);
    align-items: center;
    grid-gap: .5em;
    width: 50%;
    font-size: .55em;
    letter-spacing: .05em;
}

.insights-card-grid-title {
	grid-row: 1;
    grid-column: 1/7;
    justify-self: center;
    font-size: .85em;
    letter-spacing: .2em;
    margin-bottom: .5em;
}

.insights-card-positive-text {
	grid-row: 2;
	grid-column: 3/7;
}

.insights-card-negative-text {
	grid-row: 3;
	grid-column: 3/7;
}

.insights-card-total-text {
	grid-row: 4;
	grid-column: 3/7;
}

.insights-card-positive-score {
	grid-row: 2;
	grid-column: 1;
	justify-self: right;
}

.insights-card-negative-score {
	grid-row: 3;
	grid-column: 1;
	justify-self: right;
}

.insights-card-total-score {
	grid-row: 4;
	grid-column: 1;
	justify-self: right;
}

.insights-card-item-button-container {
	display: flex;
	width: 100%;
	justify-content: flex-start;
	height: 10%;
	align-items: center;
}

.insights-card-item-view-player-button {
    display: flex;
    margin: 1em .5em;
    border: 1px solid black;
    letter-spacing: .2em;
    font-size: .5em;
    align-items: center;
    padding: .75em 1em;
    color: black;
}



/*
// ======================================
// MEDIA QUERIES
// ======================================
*/


@media (min-width: 550px) {

    .insights-card-item-positive-negative-neutral-grid-container {
        width: 40%;
    	font-size: .65em;
    }


    

}


@media (min-width: 600px) {

    .insights-card-item-wrapper {
        width: 60vw;
    }

}



@media (min-width: 700px) {

    .insights-card-item-wrapper {
        width: 50vw;
    }

}


@media (min-width: 800px) {

    .insights-card-item-positive-negative-neutral-grid-container {
        width: 40%;
    	font-size: .65em;
    }

    .insights-card-item-wrapper {

    	width: 45vw;
    }


    .insights-card-player-shirt {
        width: 7.75em;
        height: 7.75em;
    }

}



@media (min-width: 1000px) {


    .insights-card-item-wrapper {

    	width: 35vw;
    }


}


@media (min-width: 1200px) {


    .insights-card-item-wrapper {
		height: 50vh;
    	width: 45vw;
        min-height: 425px;
    	font-size: 1.1em;
    }

    .insights-card-item-player-image-container {
        height: 25vh;
        min-height: 200px;
    }


    .insights-card-player-shirt {
        width: 9em;
        height: 9em;
    }

    .insights-card-player-name-inner {
        font-size: 2.25em;
    }


    .insights-card-item-player-name-header, .insights-card-item-rank-header {
        font-size: .85em;
    }

    .insights-card-club-name {
        
        font-size: .75em;
    }

    .insights-card-item-positive-negative-neutral-grid-container {
        font-size: .75em;
    }

    .insights-card-item-score-category {
        font-size: .85em;
    }

    .insights-card-item-view-player-button {
        font-size: .65em;
    }


}



@media (min-width: 1400px) {


    .insights-card-item-wrapper {
		height: 55vh;
        min-height: 450px;
    	width: 40vw;
    	font-size: 1.2em;
    }


    .insights-card-item-player-image-container {
        height: 30vh;
    }

}



@media (min-width: 1600px) {


    .insights-card-item-wrapper {
        width: 35vw;
    }

}



@media (min-width: 1600px) {


    .insights-card-item-wrapper {
        width: 30vw;
    }

}



