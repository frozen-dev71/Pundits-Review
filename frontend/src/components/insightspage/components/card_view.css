/*
// ======================================
// CSS for Card View Carousel on the Insights Page
// ======================================
*/


.card-view-wrapper {
	margin-top: 2.5vh;
}

.swiper-wrapper {
	padding: 1em 0;
}


/*
// ======================================
// MEDIA QUERIES
// ======================================
*/


@media (min-width: 800px) {


    .card-view-wrapper {
		margin-top: 5vh;
	}

}



@media (min-width: 1000px) {


    .card-view-wrapper {
		margin-top: 7.5vh;
	}

}