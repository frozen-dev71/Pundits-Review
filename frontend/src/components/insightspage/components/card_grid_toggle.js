// ======================================
// This file contains the CARD/GRID toggle menu
// Allows users to toggle between seeing insights in a grid or card view
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { connect } from "react-redux";

// Local JS Imports
import { setGridCardView } from '../../actions/actions';

// CSS Imports 
import './card_grid_toggle.css';

// ===========
// Card Grid Toggle Component 
// ===========


export class CardGridToggleWrapper extends Component {


	_handleClick(e) {

		if (e.target.id == "grid-button") {

			// Update the redux state with new choice
			this.props.setGridCardView("GRID");

			// Adjust the presentation
			document.querySelector('#grid-button').className = "body-text-light card-grid-button card-grid-button-active"
			document.querySelector('#card-button').className = "body-text-light card-grid-button "

		} else {

			// Update the redux state with new choice
			this.props.setGridCardView("CARD");

			// Adjust the presentation
			document.querySelector('#card-button').className = "body-text-light card-grid-button card-grid-button-active"
			document.querySelector('#grid-button').className = "body-text-light card-grid-button "

		}

	}

	render() {

		let result = (

			<div className="card-grid-toggle-wrapper">

				<div className="card-grid-button-container">

					<button id="card-button" className="body-text-light card-grid-button card-grid-button-active" onClick={(e) => this._handleClick(e)}>CARDS</button>
					<button id="grid-button" className="body-text-light card-grid-button " onClick={(e) => this._handleClick(e)}>GRID</button>

				</div>

			</div>

			);

		return result
	}
}



// ===========
// Connect component to redux store
// ===========
const mapStateToProps = state => {
  return { 

  		gridCardView: state.gridCardView

  	};
};


export const CardGridToggle = connect(mapStateToProps, {setGridCardView})(CardGridToggleWrapper);

