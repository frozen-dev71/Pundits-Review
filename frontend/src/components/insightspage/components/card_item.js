// ======================================
// CARD ITEM component - Individual container for each player in the insights page
// Rendered within Swiper carousel
// ======================================

// ===========
// IMPORTS 
// ===========

// General Imports 
import React, { Component } from "react";
import { render } from "react-dom";
import { connect } from "react-redux";
import { Link } from 'react-router-dom';

// Local JS Imports
import { club_badges } from '../../global/resources/club_badges';
import { club_shirts } from '../../global/resources/club_shirts';
import { club_colors } from '../../global/resources/club_colors';
import { setPlayer } from '../../actions/actions';

// CSS Imports 
import './card_item.css';

// ===========
// Card Item Component
// ===========

export class CardItemWrapper extends Component {

	// Set player in redux store on button click
	_handleLinkClick = () => {

		this.props.setPlayer(this.props.playerName)

	}

	render() {

		// Declare specific player page url
		let playerNameParam = "/players/" + encodeURIComponent(this.props.playerName);

		let result = (

				<div className="insights-card-item-wrapper background-light">

					<div className={"insights-card-item-header-container " + club_colors[this.props.clubName]}>

						<p className="insights-card-item-player-name-header body-text-light white-text">{this.props.playerName.toUpperCase()}</p>
						<p className="insights-card-item-rank-header body-text-light white-text">{"RANK: " + this.props.rank}</p>

					</div>

					<div className="insights-card-item-player-image-container">

						<span className={"insights-card-club-badge club-badge " + club_badges[this.props.clubName]}></span>
						<span className={"insights-card-player-shirt club-shirt " + club_shirts[this.props.clubName]}></span>
						<p className="insights-card-player-name-inner body-text">{this.props.playerName.toUpperCase()}</p>
						<p className="insights-card-club-name body-text-light white-text">{this.props.clubName.toUpperCase()}</p>


					</div>

					<div className="insights-card-item-review-score-container background-light">

						<p className="insights-card-item-score-category body-text-light">{this.props.scoreCategory.toUpperCase()}</p>
						<p className="insights-card-item-score-category-score body-text">{this.props.score}</p>

						<div className="insights-card-item-positive-negative-neutral-grid-container body-text-light">

							<p className="insights-card-grid-title">{this.props.gridTitle}</p>

							<p className="insights-card-positive-score">{this.props.positiveReviews}</p>
							<p className="insights-card-negative-score">{this.props.negativeReviews}</p>
							<p className="insights-card-total-score">{this.props.totalReviews}</p>

							<p className="insights-card-positive-text">Positive Reviews</p>
							<p className="insights-card-negative-text">Negative Reviews</p>
							<p className="insights-card-total-text">Total Reviews</p>

						</div> 

					</div>

					<div className="insights-card-item-button-container background-light"> 

						<Link to={playerNameParam} className="insights-card-item-view-player-button body-text-light background-light react-link" onClick={this._handleLinkClick}>
							VIEW PLAYER PAGE
						</Link>

					</div>

				</div>

			);

		return result;
	}
}



// ===============
// Connect redux state container to Grid Item component
// ===============


const mapStateToProps = state => {
  return { 

	  	playersViewed: state.playersViewed
  	};
};



// Export connected component
export const CardItem = connect(mapStateToProps, { setPlayer })(CardItemWrapper);