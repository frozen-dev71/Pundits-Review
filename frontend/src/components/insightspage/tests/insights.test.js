// ======================================
// This file tests for the components listed in the INSIGHTS sub-directory
// ======================================


// ===========
// TEST COVERAGE
// ===========

// 1. Check that InsightsPage renders correctly
// 2. Check that CardGridToggle renders correctly
// 3. Check that CardItem renders correctly
// 4. Check that InsightsCards renders correctly
// 5. Check that Filter renders correctly
// 6. Check that GridItem renders correctly
// 7. Check that GridTableHeader renders correctly
// 8. Check that InsightsGrid renders correctly
// 9. Check that InsightsResults renders correctly
// 10. Check that TopBottomToggle renders correctly
// 11. Check that CategoryDropdown renders correctly

// ===========
// IMPORTS 
// ===========


// General Imports
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from "react-redux";
import toJson from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';

// Local JS Imports
import { InsightsPage } from '.././insightspage';
import { CardGridToggle } from '.././components/card_grid_toggle';
import { CardItem } from '.././components/card_item';
import { InsightsCards } from '.././components/card_view';
import { CategoryDropdown } from '.././components/category_dropdown';
import { Filter } from '.././components/filter';
import { GridItem } from '.././components/grid_item';
import { GridTableHeader } from '.././components/grid_table_header';
import { InsightsGrid } from '.././components/grid_view';
import { InsightsResults } from '.././components/insights_results';
import { TopBottomToggle } from '.././components/top_bottom_toggle';

import store from '../../store/store';

// Configurations
Enzyme.configure({ adapter: new Adapter() })

// ===========
// Suppression of SSL Warning 
//
// Code adapted from comment from sketchbuch on GitHub issues post 'Warning: useLayoutEffect does nothing on the server #1373' on 12 Oct 2019
// https://github.com/reduxjs/react-redux/issues/1373
// Accessed 09 Sep 2020
//
// __Note__:
//	
//		- Jest produces "Warning: useLayoutEffect does nothing on the server" & 'Warning: React.createElement: type is invalid -- expected a string (for built-in components)' - warning which confuses test cases 
// 		- Wrapper test suppresses that warning for clear test reports
// ===========

describe('Insights Page Component Tests', () => {
  	const originalConsoleError = console.error;

  	beforeEach(() => {
    	console.error = jest.fn((msg) => {
	      	if (msg.includes('Warning: useLayoutEffect does nothing on the server') || msg.includes('Warning: React.createElement: type is invalid -- expected a string (for built-in components)')) {
	       		return null;
	      	} else {
	        	originalConsoleError(msg);
	      	}
    	});
  	});

  afterEach(() => {
    console.error = originalConsoleError;
  });


// End of referenced code

  	// ===========
	// TEST 1. Check that InsightsPage renders correctly
	// ===========

	describe('InsightsPage', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><InsightsPage/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 2. Check that CardGridToggle renders correctly
	// ===========

	describe('CardGridToggle', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><CardGridToggle/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 3. Check that CardItem renders correctly
	// ===========

	describe('CardItem', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><CardItem/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = shallow(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 4. Check that InsightsCards renders correctly
	// ===========

	describe('InsightsCards', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><InsightsCards/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 5. Check that Filter renders correctly
	// ===========

	describe('Filter', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><Filter/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 6. Check that GridItem renders correctly
	// ===========

	describe('GridItem', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><GridItem/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 7. Check that GridTableHeader renders correctly
	// ===========

	describe('GridTableHeader', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><GridTableHeader/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 8. Check that InsightsGrid renders correctly
	// ===========

	describe('InsightsGrid', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><InsightsGrid/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 9. Check that InsightsResults renders correctly
	// ===========

	describe('InsightsResults', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><InsightsResults/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


	// ===========
	// TEST 10. Check that TopBottomToggle renders correctly
	// ===========

	describe('TopBottomToggle', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><TopBottomToggle/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})



	// ===========
	// TEST 11. Check that InsightsCards renders correctly
	// ===========

	describe('CategoryDropdown', () => {

	  	it('should render correctly', () => {

		  	// Declare component
		  	const component = (<BrowserRouter><Provider store={store}><CategoryDropdown/></Provider></BrowserRouter>);
			
			//
			// Simulate Component
		    const wrapper = render(component);

		    // Run Test Case
		    expect(toJson(wrapper)).toMatchSnapshot();

	  	});

	})


});
