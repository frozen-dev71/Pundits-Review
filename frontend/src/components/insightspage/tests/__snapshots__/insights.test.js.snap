// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Insights Page Component Tests CardGridToggle should render correctly 1`] = `
<div
  class="card-grid-toggle-wrapper"
>
  <div
    class="card-grid-button-container"
  >
    <button
      class="body-text-light card-grid-button card-grid-button-active"
      id="card-button"
    >
      CARDS
    </button>
    <button
      class="body-text-light card-grid-button "
      id="grid-button"
    >
      GRID
    </button>
  </div>
</div>
`;

exports[`Insights Page Component Tests CardItem should render correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(CardItemWrapper) />
  </Provider>
</Router>
`;

exports[`Insights Page Component Tests CategoryDropdown should render correctly 1`] = `
<div
  class="category-dropdown-container background-dark"
>
  <div
    class="category-key-container"
  >
    <p
      class="category-key-text body-text-light white-text"
    >
      CATEGORY:
    </p>
  </div>
  <div
    class="category-dropdown-menu-wrapper white-text body-text-light"
  >
    <ul
      class="custom-select__trigger"
    >
      <li
        class="body-text-light current-select-option"
        id="current-category"
        role="button"
        tabindex="0"
      >
        REVIEW SCORE
      </li>
      <div
        class="arrow"
      />
    </ul>
    <ul
      class="custom-options"
      role="list"
    >
      <li
        class="custom-option background-dark"
        id="review-score-dropdown"
        role="button"
        tabindex="0"
      >
        <span
          class="body-text-light select-option"
        >
          REVIEW SCORE
        </span>
      </li>
      <li
        class="custom-option background-dark "
        id="season-avg-score-dropdown"
        role="button"
        tabindex="0"
      >
        <span
          class="body-text-light select-option"
        >
          SEASON AVG SCORE
        </span>
      </li>
      <li
        class="custom-option background-dark "
        id="mentions-dropdown"
        role="button"
        tabindex="0"
      >
        <span
          class="body-text-light select-option"
        >
          MENTIONS
        </span>
      </li>
      <li
        class="custom-option background-dark "
        id="positive-score-dropdown"
        role="button"
        tabindex="0"
      >
        <span
          class="body-text-light select-option"
        >
          POSITIVE REVIEWS
        </span>
      </li>
      <li
        class="custom-option background-dark "
        id="negative-score-dropdown"
        role="button"
        tabindex="0"
      >
        <span
          class="body-text-light select-option"
        >
          NEGATIVE REVIEWS
        </span>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Insights Page Component Tests Filter should render correctly 1`] = `
<div
  class="filter-button-container"
>
  <button
    class="filter-button body-text-light"
  >
    FILTER
  </button>
</div>
`;

exports[`Insights Page Component Tests GridItem should render correctly 1`] = `
<a
  class="react-link grid-item-wrapper grid-item-wrapper-white"
  href="/players/undefined"
  role="button"
>
  <p
    class="body-text-light grid-item-text grid-item-rank grid-item-border-grey"
  />
  <p
    class="body-text-light grid-item-text grid-item-playername"
  />
  <div
    class="body-text-light grid-item-text grid-item-clubname grid-item-border-grey"
  >
    <span
      class="undefined club-badge grid-item-club-badge"
    />
    <span />
  </div>
  <p
    class="body-text-light grid-item-text grid-item-score"
  />
</a>
`;

exports[`Insights Page Component Tests GridTableHeader should render correctly 1`] = `
<div
  class="grid-table-header-wrapper body-text-light white-text"
>
  <p
    class="grid-table-column-headers rank-header"
  >
    RANK
  </p>
  <p
    class="grid-table-column-headers player-header"
  >
    PLAYER
  </p>
  <p
    class="grid-table-column-headers club-header"
  >
    CLUB
  </p>
  <p
    class="grid-table-column-headers score-header"
  />
</div>
`;

exports[`Insights Page Component Tests InsightsCards should render correctly 1`] = `
<div
  class="card-view-wrapper"
>
  <div
    class="swiper-container"
  >
    <div
      class="swiper-button-prev"
    />
    <div
      class="swiper-button-next"
    />
    <div
      class="swiper-wrapper"
    />
  </div>
</div>
`;

exports[`Insights Page Component Tests InsightsGrid should render correctly 1`] = `
<div
  class="loading-screen-wrapper"
>
  <p
    class="loading-screen-text body-text-light"
  >
    LOADING
  </p>
</div>
`;

exports[`Insights Page Component Tests InsightsPage should render correctly 1`] = `
<div
  class="insights-page-wrapper"
>
  <p
    class="insights-page-title heading-text"
  >
    Player Rankings
  </p>
  <div
    class="insights-top-bottom-control-wrapper"
  >
    <div
      class="top-bottom-toggle-wrapper"
    >
      <div
        aria-required="true"
        class="top-bottom-inner-container insights-toggle-active"
        id="top-toggle"
        role="button"
        tabindex="0"
      >
        <p
          class="top-bottom-menu-text heading-text"
        >
          Top
        </p>
      </div>
      <div
        aria-required="true"
        class="top-bottom-inner-container"
        id="bottom-toggle"
        role="button"
        tabindex="0"
      >
        <p
          class="top-bottom-menu-text heading-text"
        >
          Bottom
        </p>
      </div>
    </div>
    <div
      class="category-dropdown-container background-dark"
    >
      <div
        class="category-key-container"
      >
        <p
          class="category-key-text body-text-light white-text"
        >
          CATEGORY:
        </p>
      </div>
      <div
        class="category-dropdown-menu-wrapper white-text body-text-light"
      >
        <ul
          class="custom-select__trigger"
        >
          <li
            class="body-text-light current-select-option"
            id="current-category"
            role="button"
            tabindex="0"
          >
            REVIEW SCORE
          </li>
          <div
            class="arrow"
          />
        </ul>
        <ul
          class="custom-options"
          role="list"
        >
          <li
            class="custom-option background-dark"
            id="review-score-dropdown"
            role="button"
            tabindex="0"
          >
            <span
              class="body-text-light select-option"
            >
              REVIEW SCORE
            </span>
          </li>
          <li
            class="custom-option background-dark "
            id="season-avg-score-dropdown"
            role="button"
            tabindex="0"
          >
            <span
              class="body-text-light select-option"
            >
              SEASON AVG SCORE
            </span>
          </li>
          <li
            class="custom-option background-dark "
            id="mentions-dropdown"
            role="button"
            tabindex="0"
          >
            <span
              class="body-text-light select-option"
            >
              MENTIONS
            </span>
          </li>
          <li
            class="custom-option background-dark "
            id="positive-score-dropdown"
            role="button"
            tabindex="0"
          >
            <span
              class="body-text-light select-option"
            >
              POSITIVE REVIEWS
            </span>
          </li>
          <li
            class="custom-option background-dark "
            id="negative-score-dropdown"
            role="button"
            tabindex="0"
          >
            <span
              class="body-text-light select-option"
            >
              NEGATIVE REVIEWS
            </span>
          </li>
        </ul>
      </div>
    </div>
    <div
      class="filter-button-container"
    >
      <button
        class="filter-button body-text-light"
      >
        FILTER
      </button>
    </div>
  </div>
  <div
    class="card-grid-toggle-wrapper"
  >
    <div
      class="card-grid-button-container"
    >
      <button
        class="body-text-light card-grid-button card-grid-button-active"
        id="card-button"
      >
        CARDS
      </button>
      <button
        class="body-text-light card-grid-button "
        id="grid-button"
      >
        GRID
      </button>
    </div>
  </div>
  <div
    class="card-view-wrapper"
  >
    <div
      class="swiper-container"
    >
      <div
        class="swiper-button-prev"
      />
      <div
        class="swiper-button-next"
      />
      <div
        class="swiper-wrapper"
      />
    </div>
  </div>
  <div
    class="footer-wrapper"
  >
    <p
      class="footer-title body-text-light"
    >
      GET IN TOUCH
    </p>
    <div
      class="footer-logo-wrapper"
    >
      <a
        class="react-link"
        href="https://twitter.com/AndyClarkeMedia"
        target="_blank"
      >
        <i
          aria-hidden="true"
          class="fa fa-twitter media-icon"
        />
      </a>
      <a
        class="footer-logo heading-text react-link"
        href="/"
      >
        PR
      </a>
      <a
        class="react-link"
        href="mailto:clarkeAJ3@cardiff.ac.uk"
        target="_blank"
      >
        <i
          aria-hidden="true"
          class="fa fa-envelope media-icon"
        />
      </a>
    </div>
    <p
      class="footer-copyright-notice body-text-light"
    >
      © Andy Clarke, 2020
    </p>
  </div>
</div>
`;

exports[`Insights Page Component Tests InsightsResults should render correctly 1`] = `
<div
  class="card-view-wrapper"
>
  <div
    class="swiper-container"
  >
    <div
      class="swiper-button-prev"
    />
    <div
      class="swiper-button-next"
    />
    <div
      class="swiper-wrapper"
    />
  </div>
</div>
`;

exports[`Insights Page Component Tests TopBottomToggle should render correctly 1`] = `
<div
  class="insights-top-bottom-control-wrapper"
>
  <div
    class="top-bottom-toggle-wrapper"
  >
    <div
      aria-required="true"
      class="top-bottom-inner-container insights-toggle-active"
      id="top-toggle"
      role="button"
      tabindex="0"
    >
      <p
        class="top-bottom-menu-text heading-text"
      >
        Top
      </p>
    </div>
    <div
      aria-required="true"
      class="top-bottom-inner-container"
      id="bottom-toggle"
      role="button"
      tabindex="0"
    >
      <p
        class="top-bottom-menu-text heading-text"
      >
        Bottom
      </p>
    </div>
  </div>
  <div
    class="category-dropdown-container background-dark"
  >
    <div
      class="category-key-container"
    >
      <p
        class="category-key-text body-text-light white-text"
      >
        CATEGORY:
      </p>
    </div>
    <div
      class="category-dropdown-menu-wrapper white-text body-text-light"
    >
      <ul
        class="custom-select__trigger"
      >
        <li
          class="body-text-light current-select-option"
          id="current-category"
          role="button"
          tabindex="0"
        >
          REVIEW SCORE
        </li>
        <div
          class="arrow"
        />
      </ul>
      <ul
        class="custom-options"
        role="list"
      >
        <li
          class="custom-option background-dark"
          id="review-score-dropdown"
          role="button"
          tabindex="0"
        >
          <span
            class="body-text-light select-option"
          >
            REVIEW SCORE
          </span>
        </li>
        <li
          class="custom-option background-dark "
          id="season-avg-score-dropdown"
          role="button"
          tabindex="0"
        >
          <span
            class="body-text-light select-option"
          >
            SEASON AVG SCORE
          </span>
        </li>
        <li
          class="custom-option background-dark "
          id="mentions-dropdown"
          role="button"
          tabindex="0"
        >
          <span
            class="body-text-light select-option"
          >
            MENTIONS
          </span>
        </li>
        <li
          class="custom-option background-dark "
          id="positive-score-dropdown"
          role="button"
          tabindex="0"
        >
          <span
            class="body-text-light select-option"
          >
            POSITIVE REVIEWS
          </span>
        </li>
        <li
          class="custom-option background-dark "
          id="negative-score-dropdown"
          role="button"
          tabindex="0"
        >
          <span
            class="body-text-light select-option"
          >
            NEGATIVE REVIEWS
          </span>
        </li>
      </ul>
    </div>
  </div>
  <div
    class="filter-button-container"
  >
    <button
      class="filter-button body-text-light"
    >
      FILTER
    </button>
  </div>
</div>
`;
